actors/builtin/reward/reward_state.go

```diff
@@ -103,20 +105,21 @@ func (st *State) withdrawReward(store adt.Store, owner addr.Address, currEpoch a
 	return withdrawableSum, nil
 }
 
+// AmountVested returns the `TokenAmount` value of funds vested in the reward at an epoch
 func (r *Reward) AmountVested(currEpoch abi.ChainEpoch) abi.TokenAmount {
-	elapsed := currEpoch - r.StartEpoch
 	switch r.VestingFunction {
 	case None:
 		return r.Value
 	case Linear:
-		vestDuration := big.Sub(big.NewInt(int64(r.EndEpoch)), big.NewInt(int64(r.StartEpoch)))
-		if big.NewInt(int64(elapsed)).GreaterThanEqual(vestDuration) {
+		elapsed := currEpoch - r.StartEpoch
+		vestDuration := r.EndEpoch - r.StartEpoch
+		if elapsed >= vestDuration {
 			return r.Value
 		}
 
 		// totalReward * elapsedEpoch / vestDuration
-		return big.Div(big.Mul(r.Value, big.NewInt(int64(elapsed))), vestDuration)
+		return big.Mul(r.Value, big.NewInt(int64(elapsed/vestDuration)))
```

Comment:
This changes the behaviour incorrectly. `elapsed/vestDuration` as a  real number is always < 1 here, which means this integer division will truncate to zero. The division must be performed as the _last_ operation in the expression. Deferring the bigint allocation until here is good though, thanks.

This is a general guideline for all integer arithmetic in the project. But since it tripped up at least one person here, would you add a comment explaining it to the next reader?

Reply:
ah, yes, a lapse on my part. I'll fix that ordering now